.github/workflows/tests.yml

```
@@ -0,0 +1,30 @@
+# Authored by Alexandra N. Walker <alex.walker@indicio.tech> ☀️
+
+name: Tests
+on:
+  push:
+    branches: 
+      - main
+      - feature/revocation
+      - test/tests
+  pull_request:
+    branches:
+      - main
+      - feature/revocation
+
+env:
+  POETRY_VERSION: 1.1.11
+
+jobs:
+  test:
+  name: Tests
+  runs-on: ubuntu-latest
+  strategy:
+    matrix:
+      python-version: [3.6, 3.7, 3.8, 3.9]
+
+  steps:
+  - uses: actions/checkout@v2
+  - name: Test with pytest
+    run: |
+      poetry run pytest
```

We need to run the tests in conjunction with the agents started up in the docker compose. The simplest way to do that is:
```suggestion
  - name: Test with docker-compose
    run: |
      docker-compose run tests
```